specs/prod-genomes.spec

```diff
@@ -186,190 +211,298 @@ module KBaseGenomes {
+        mapping<string ontology_namespace,mapping<string ontology_id,list<int> evidence_event>> ontology_terms;
+        string note;
+        string md5;
+        string parent_gene;
+        list<string> children;
+        list<string> flags;
+        list<string> warnings;
+        list <InferenceInfo> inference_data;
+        string dna_sequence;
+        int dna_sequence_length;
+        list<tuple<string fieldname,string alias>> aliases;
+        list<tuple<string db_source,string db_identifier>> db_xrefs;
     } NonCodingFeature;
 
-
-        /*
-          Structure for a single feature CDS
-
-      flags are flag fields in GenBank format. This will be a controlled vocabulary.
-        Initially Acceptable values are pseudo, ribosomal_slippage, and trans_splicing
-        Md5 is the md5 of dna_sequence.
-
-        @optional parent_gene parent_mrna functions ontology_terms note flags warnings
-        @optional inference_data dna_sequence aliases db_xrefs functional_descriptions
+    /*
+    Structure for a single coding sequence.
+
+    Coding sequences are the sections of a feature's sequence that are translated
+    to a protein (minus introns and UTRs).
+
+    Fields:
+        id - string - identifier of the coding sequence, such as "b0001_CDS_1"
+        location - list<tuple<string, int, string, int>> - list of
+            locations from where this sequence originates in the original assembly.
+            Each sub-sequence in the list constitutes a section of the resulting
+            CDS. The first element in the tuple corresponds to the "contig_id",
+            such as "NC_000913.3". The second element in the tuple is an index in
+            the contig of where the sequence starts. The third element is either a
+            plus or minus sign indicating whether it is on the 5' to 3' leading
+            strand ("+") or on the 3' to 5' lagging strand ("-"). The last element
+            is the length of the sub-sequence.
+            For a location on the leading strand (denoted by "+"), the index is
+            of the leftmost base, and the sequence extends to the right. For a
+            location on the lagging strand (denoted by "-"), the index is of
+            the rightmost base, and the sequence extends to the left.
+            NOTE: the last element in each tuple is the *length* of each
+            sub-sequence. If you have a location such as ("xyz", 100, "+", 50),
+            then your sequence will go from index 100 to index 149 (this has a
+            length of 50). It *does not* go from index 100 to index 150, as
+            that would have a length of 51.
+            Likewise, if you have the location ("xyz", 100, "-", 50), then the
+            sequence extends from 100 down to 51, which has a length of 50
+            bases. It does not go from index 100 to 50, as that would have a
+            length of 51.
+        md5 - string - md5 of the dna sequence - TODO clarification
```

Comment:
I think only thing to add here is something like "useful for quickly comparing if two sequences are identical"